views/js/smartsupp.js

```diff
@@ -106,36 +124,53 @@ jQuery(document).ready( function($) {
             },
             dataType: 'json',
             headers: { "cache-control": "no-cache" },
-            success: function(data) {
-                $("input#smartsupp_key").val(data.key);
-                    $("#smartsupp_configuration p.email").html(data.email);
-                    if (data.error === null) {
-                        $("#smartsupp_create_account .alerts").hide();
-                    }
-                    else {
-                        $("#smartsupp_create_account .alerts").show();
-                        $("#smartsupp_create_account .alerts .alert").html(data.message);
+            success: function (response) {
+                if (response.error) {
+                    $("#smartsupp_create_account .alerts").show();
+
+                    if (response.message) {
+                        $("#smartsupp_create_account .alerts .alert").html(response.message);
                     }
+
+                    return;
+                }
+
+                $("#smartsupp_create_account .alerts .alert").html(smartsupp.genericAjaxErrorMessage);
+            },
+            error: function (response) {
+                console.error(response);
+
+                if (smartsupp.genericAjaxErrorMessage) {
+                    $("#smartsupp_create_account .alerts").show();
+                    $("#smartsupp_create_account .alerts .alert").html(smartsupp.genericAjaxErrorMessage);
+                }
             }
         });        
         page_refresh();    
     });
         
     $( "#deactivate_chat_do" ).click(function() {
         $("#smartsupp_configuration").next('.bootstrap').hide();
+
         $.ajax({
-                url: ajax_controller_url,
-                async: false,
-                type: 'POST',
-                data: {
-                    action: 'deactivate'
-                },
-                dataType: 'json',
-                headers: { "cache-control": "no-cache" },
-                success: function(data) {
-                        $("input#smartsupp_key").val(data.key);
-                        $("#smartsupp_configuration p.email").html(data.email);
-                }
+            url: ajax_controller_url,
+            async: false,
+            type: 'POST',
+            data: {
+                action: 'deactivate'
+            },
+            dataType: 'json',
+            headers: { "cache-control": "no-cache" },
+            success: function (response) {
+                $("input#smartsupp_key").val(response.key);

```

Comment:
here we won't have any errors in response body? If they exist, add the same alert block. Ideally, could just adapt how errors are showed above and reuse same blocks. You didn't post any images so not sure how these validations looks like

Reply:
this ajax only removes values from the configuration tab. So no errors or messages are parsed as a response. No api calls either.